### PR TITLE
CMake fixes

### PR DESCRIPTION
**Pull request by Ninja3047**
* remove thirdparty_ interfaces
* undo llvm jank
* update git_watcher.cmake, fix quoting
* update FindZ3.cmake, remove testz3.cpp (now embedded in the cmake)

it compiles but may have unintentially broke things

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -255,9 +205,14 @@ target_compile_definitions(remill_settings INTERFACE
   "REMILL_BUILD_SEMANTICS_DIR_SPARC64=\"${REMILL_BUILD_SEMANTICS_DIR_SPARC64}\""
 )
 
-set(THIRDPARTY_LIBRARY_LIST thirdparty_llvm thirdparty_xed thirdparty_glog thirdparty_gflags thirdparty_sleigh)
 target_link_libraries(remill_settings INTERFACE
```
Comment (Ninja3047) [inline CMakeLists.txt:208]: should deps go into _settings as well?

File: cmake/remillConfig.cmake.in [hunk 2]
```diff
@@ -48,28 +48,12 @@ if(NOT remill_setup_vcpkg AND NOT TARGET remill)
...
-    list(REMOVE_ITEM LLVMSupport_LIBS Z3)
-    list(APPEND LLVMSupport_LIBS z3::libz3)
-    set_target_properties(LLVMSupport PROPERTIES
-      INTERFACE_LINK_LIBRARIES "${LLVMSupport_LIBS}")
-  endif()
 
   # Exported Targets
   include("${CMAKE_CURRENT_LIST_DIR}/remillTargets.cmake")
 
-  if(TARGET thirdparty_llvm)
```
Comment (Ninja3047) [inline cmake/remillConfig.cmake.in:65]: unsure what this was doing so i just deleted it and it didn't appear to break anything

File: CMakeLists.txt (hunk 1 above)
Comment (ekilmer) [reply CMakeLists.txt:208]: It depends on how precise you want to be with the linking of the dependencies to the various libraries/executables.

Ideally, you only link to the dependencies you need, but adding them here means you only need to change one spot 🤷

**Review by ekilmer — APPROVED** [review]
LGTM!

**Pull request merged**